### ISSUE TITLE
Setting PublicOrigin changes the RedirectUrl to the same Scheme

### ISSUE THREAD

**Issue by schwarzie2478** [IdentityServer/IdentityServer4#2018]
### Issue / Steps to reproduce the problem

IdentityServer is running on server behind a loadbalancer.
The Load balancer takes care of the https protocol.
Our application server doesn't know about it.

As shown in the article 
[https://leastprivilege.com/2017/10/09/new-in-identityserver4-v2-simplified-configuration-behind-load-balancers-or-reverse-proxies/](url)  
I set the PublicOrigin to the https url.

Now the well-known configuration reflects the url as you should call it from outside.

My client website is hosted on a  http:// location  ( not secure: hosted inside the company)
I have registered a client with an implicit flow and an Redirect Url in the format of http://xxxx.com/signed_in?

but the response from IdentityServer  gives the redirect url  https://xxxx.com/signed_in? 

I think that somewhere the Response.RedirectUrl is copied from the Request.RedirectUrl, but using the Scheme determined from the Request which has been modified by parsing the PublicOrigin in the BaseUrlMiddleware.

I haven't found a place where I can change the Url back to the one that was in the request.

**Comment by MentorMateYosif** [comment]
Hello,
You can create middle ware that rewrites the HttpContext.Request.Scheme
By monitoring the "X-Forwarded-Proto" header value.
Regards,
Yosif

**Comment by schwarzie2478** [comment]
Hey,

I checked the headers I recieve in my request and that one is not there.
I do have X-Origin-Proto but that pointed to the site inside server I'm hosted on.

I also tried to adding a Use block with the following code
`        public void Configure(IApplicationBuilder app)
        {
            app.Use(async (context, next) =>
            {
                context.Request.Scheme ="https";
                await next.Invoke();
            });`

Is it possible to modify the Response somehow after next.Invoke() returns?

**Comment by schwarzie2478** [comment]
I managed to go through the entire login flow and I documented the Headers in the Request and Reponse objects when entering and exiting IdentityServer.

When I log the headers for the response which redirects back to the client,  the url is still the orignal.

So I must conclude this change is not happening inside IdentityServer as I suspected but this is done probable by the reverse proxy as a rule. 

This issue is 'solved' then :-)
